### PR TITLE
Update rust.json

### PR DESCRIPTION
**Pull request by mark-i-m**
cc @matklad @Mark-Simulacrum 

I think these are the correct changes, but I'm not super sure.

### CODE REVIEW

File: highfive/configs/rust-lang/rust.json [hunk 1]
```diff
@@ -24,13 +24,13 @@
...
         "library/proc_macro":        ["@petrochenkov"],
-        "src/librustc":             ["compiler-team", "compiler-team-contributors"],
+        "compiler/rustc_middle":     ["compiler-team", "compiler-team-contributors"],
         "src/librustdoc":           ["rustdoc"],
         "library/std":               ["libs"],
         "library/term":              ["libs"],
         "library/test":              ["libs"],
         "src/llvm-project":         ["@cuviper"],
-        "src/rustc":                ["compiler-team", "compiler-team-contributors"],
+        "compiler/rustc":            ["compiler-team", "compiler-team-contributors"],
```
Comment (matklad) [inline highfive/configs/rust-lang/rust.json:33]: Yeah, this feels weird... I would expect this to be just compiler/

Comment (Mark-Simulacrum) [reply highfive/configs/rust-lang/rust.json:33]: This used to be src/rustc as a prefix for `src/rustc_*` I suspect, so this is probably right.

The librustc prefix was previously needed but we now longer have lib prefixes so we can drop it I think.

Comment (mark-i-m) [reply highfive/configs/rust-lang/rust.json:33]: I'm pretty confused now... should I drop line 27 altogether, and just leave this line, then?

Comment (Mark-Simulacrum) [reply highfive/configs/rust-lang/rust.json:33]: That would be my expectation, yes.

Comment (Mark-Simulacrum) [reply highfive/configs/rust-lang/rust.json:33]: We can probably just say "compiler" without the rustc prefix as well.

**Comment by Mark-Simulacrum** [comment]
Hopefully highfive assigns correctly with this merged, but we can iterate if needed.

**Pull request merged**